envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -62,36 +62,40 @@ end
 
 function envoy_on_response(handle)
     local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac") or {}
-    local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
-    local allowed_client = lua_metadata["request.info.allowed_client"] or false
-
-    if should_log_request(rbacMetadata, allowed_client) then
-        log_request(lua_metadata, handle)
-    end
-end
-
-function should_log_request(rbacMetadata, allowed_client)
     local is_shadow_denied = (rbacMetadata["shadow_engine_result"] or "") == "denied"
+    local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
```

Comment:
can be moved after `if is_shadow_denied then`. The less we do for 99% not requests, the better :)